src/crypto/tls/handshake_client_tls13.go

```diff
@@ -67,6 +81,21 @@ func (hs *clientHandshakeStateTLS13) handshake() error {
 		}
 	}
 
+	// Determine if ECH was accepted.
+	if hs.c.ech.status.Offered {
+		secret := hs.suite.extract(hs.helloInner.random, nil)
+		context := hs.serverHello.random[:24]
+		conf := hs.suite.expandLabel(secret, echTls13LabelAcceptConfirm, context, 8)
+		if bytes.Equal(hs.serverHello.random[24:], conf) {
+			c.ech.status.Accepted = true
```

Comment:
Can you clarify what you mean? Do you mean if acceptance is not indicated (client presumes rejection), then there are client behaviors that may not be adhered to currently?

Reply:
Sure: if ECH is rejected, servers can't send a CertificateRequest. Do we have checks in place right now that prevent this from happening?

Reply:
I see. These checks don't exist. I still haven't done a fresh read through of the spec, which is why I missed this. I'll get to this Monday.

Reply:
The spec is silent about the server's CertificateRequest behavior. The only text I could find about this is [here](https://tlswg.org/draft-ietf-tls-esni/draft-ietf-tls-esni.html#section-6.3.2.1):
> If the server requests a client certificate, the client MUST respond with an empty Certificate message, denoting no client certificate.

Enforcing this "MUST" seems pretty invasive, since the rejection code path may use any TLS version that both the client and server support. Not only do we need to enforce it in the 1.3 handshake but also in 1.2, 1.1, and so on. Is there a good reason why this should be enforced? It seems to me that a "MAY" would be safe, since the client is supposed to abort the handshake anyway.

Reply:
> The spec is silent about the server's CertificateRequest behavior. 

Is it? You cited the same text I cited, which says precisely what needs to be enforced, I think.

> Enforcing this "MUST" seems pretty invasive, since the rejection code path may use any TLS version that both the client and server support. Not only do we need to enforce it in the 1.3 handshake but also in 1.2, 1.1, and so on. Is there a good reason why this should be enforced? It seems to me that a "MAY" would be safe, since the client is supposed to abort the handshake anyway.

Sure: clients should not offer up any sensitive information to an entity that is *not* the origin without good reason. I think we can implement that enforcement in the future (and flag it as an open issue or bug now), but we certainly shouldn't change the spec.

Reply:
> > The spec is silent about the server's CertificateRequest behavior.
> 
> Is it? You cited the same text I cited, which says precisely what needs to be enforced, I think.

It seems to me that the quoted text specifies client behavior, but does not imply server behavior. I could be wrong, however.

> Sure: clients should not offer up any sensitive information to an entity that is _not_ the origin without good reason. I think we can implement that enforcement in the future (and flag it as an open issue or bug now), but we certainly shouldn't change the spec.

Ah, I see. I'll document this as a BUG and leave it for later.

Reply:
> It seems to me that the quoted text specifies client behavior, but does not imply server behavior. I could be wrong, however.

I guess the server behavior that's implied here is that the server is not allowed to **require** mutual authentication when ECH is rejected. Is that what you mean?

Reply:
Huh, maybe I was looking at older editor text, but the draft seems pretty clear: if the server sends a CertificateRequest in this case, clients must reply with an empty Certificate.

Reply:
@cjpatton is this one resolved (even if with a TODO for later)?